packages/root-cms/ui/pages/DocumentPage/DocumentPage.tsx

```diff
@@ -342,35 +342,48 @@ DocumentPage.Preview = (props: PreviewProps) => {
   }
 
   useEffect(() => {
-    if (device === '') {
+    function updateIframeStyle() {
+      if (device === '') {
+        setIframeStyle({
+          '--iframe-width': '100%',
+          '--iframe-height': '100%',
+          '--iframe-scale': '1',
+        });
+        return;
+      }
+
+      const iframe = iframeRef.current!;
+      const container = iframe.parentElement!.parentElement as HTMLElement;
+      const rect = container.getBoundingClientRect();
+      const [width, height] = DeviceResolution[device];
+      const padding = 20;
+      let scale = 1;
+      if (
+        width > rect.width - 2 * padding ||
+        height > rect.height - 2 * padding
+      ) {
+        scale = Math.min(
+          (rect.width - 2 * padding) / width,
+          (rect.height - 2 * padding) / height
+        );
+      }
       setIframeStyle({
-        '--iframe-width': '100%',
-        '--iframe-height': '100%',
-        '--iframe-scale': '1',
+        '--iframe-width': `${width}px`,
+        '--iframe-height': `${height}px`,
+        '--iframe-scale': String(scale),
       });
-      return;
     }
 
-    const iframe = iframeRef.current!;
-    const container = iframe.parentElement!.parentElement as HTMLElement;
-    const rect = container.getBoundingClientRect();
-    const [width, height] = DeviceResolution[device];
-    const padding = 20;
-    let scale = 1;
-    if (
-      width > rect.width - 2 * padding ||
-      height > rect.height - 2 * padding
-    ) {
-      scale = Math.min(
-        (rect.width - 2 * padding) / width,
-        (rect.height - 2 * padding) / height
-      );
-    }
-    setIframeStyle({
-      '--iframe-width': `${width}px`,
-      '--iframe-height': `${height}px`,
-      '--iframe-scale': String(scale),
-    });
+    updateIframeStyle();
+
+    // Maintain the aspect ratio when the window is resized.
+    // Without this, the iframe gets cut off.
+    const handleResize = () => updateIframeStyle();
```

Comment:
can you debounce this?

Reply:
hmm let's go back to without the debounce, eric used to say css transitions create artificial latency and i'm feeling a little bit of that here. reason i suggested debouncing is to reduce cpu load since we have some perf problems at the moment anything we can do to help reduce unresponsiveness the better.

Reply:
got it - yeah, there are tradeoffs with both. i do like how snappy and immediate it is _without_ the transitions so i'm aligned to that. i read a post the other day about how disabling transitions on macOS makes the system feel "faster" so i think this update is aligned to that feeling